### PR TITLE
Hotfix: Remove Lodash as Dependency

### LINKED ISSUE #473: Remove Dependency on Lodash
# User Story

> As a developer, I want to remove the dependency on lodash to avoid an issue where lodash needs to be included in dependencies in userland code to avoid a fatal.

# Acceptance Criteria

- Find and replace instances of code in `block-editor-tools` that depends on lodash. As of the time of writing, this should be limited to the use of `cloneDeep` in `usePostMeta`.

# Description of the bug

SlotFill fields stopped working after updating to WordPress 6.4.1

Upon typing in text or attempting to make changes to editing text into field nothing happens. I'd expect to be able to add/edit text. 

## Steps To Reproduce

1. Ensure the current WordPress version is `6.4.1`
2. Add the hook `usePostMetaValue` to get a post meta value
3. Create a slotfill and TextControl to manage that post meta value
4. Attempt to edit the TextControl field from the SlotFill.

## Additional Information

I noticed this JS error in the browser console:
`Uncaught TypeError: Cannot read properties of undefined (reading 'cloneDeep')`

It appears that Lodash is missing in the window object and the Lodash library is not being loaded. See https://github.com/WordPress/gutenberg/pull/52571

### PR DESCRIPTION
**Pull request by efuller**
Fixes #473 

## Summary by CodeRabbit

- **Refactor**
  - Optimized the post meta handling to improve performance and reduce dependencies.

- **Documentation**
  - Updated the package documentation to reflect the removal of `lodash` as a dependency.

### CODE REVIEW

File: packages/block-editor-tools/src/hooks/use-post-meta/index.js [hunk 1]
```diff
@@ -31,16 +30,10 @@ const usePostMeta = (postType = null, postId = null) => {
...
-   * objects or array values within meta keys not triggering React or
-   * Gutenberg's state management system realizing that there is a change due
-   * to the fact that sub-items are stored as object references. These bugs are
-   * extremely difficult to find and correct, so it makes sense to include this
-   * functionality here as a catch-all on updates.
+   * Define a wrapper for the setMeta function that spreads the next meta value into a new object.
    * @param {object} next - The new value for meta.
    */
-  const setMetaSafe = (next) => setMeta(cloneDeep(next));
+  const setMetaSafe = (next) => setMeta({ ...next });
```
[the remark this answers is not in the document]
Comment (efuller) [reply packages/block-editor-tools/src/hooks/use-post-meta/index.js:36]: For now, we will assume that an object spread will suffice.

**Comment by stevenslack** [comment]
@efuller Can you add an entry in the [changelog](https://github.com/alleyinteractive/alley-scripts/blob/main/packages/block-editor-tools/README.md#changelog)? it is currently located in the README.md for the `block-editor-tools`

**Comment by efuller** [comment]
@stevenslack updated with ba7fab8

**Review by stevenslack — APPROVED** [review]
Looks good 🌴

**Pull request merged**